File: app/src/main/java/fr/ekito/myweatherapp/WeatherFormat.java
package fr.ekito.myweatherapp;

import android.support.annotation.NonNull;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import fr.ekito.myweatherlibrary.json.weather.Forecastday_;

/**
 * Created by arnaud on 04/08/2016.
 */
public class WeatherFormat {
    public static List<Forecastday_> filterForecast(List<Forecastday_> forecastday) {
        List<Forecastday_> filtered = new ArrayList<>();
        System.out.println("");
        for (Forecastday_ f : forecastday) {
            if (!f.getIcon().startsWith("nt_")) {
                filtered.add(f);
                if (filtered.size() >= 4) {
                    break;
                }
            }
        }
        return filtered;
    }

    @NonNull
    public static String getTemp(Forecastday_ day1) {
        return day1.getLow().getCelsius() + " - " + day1.getHigh().getCelsius();
    }

    public static void displayWeatherIcon(TextView txt, int size, Forecastday_ f) {
        txt.setTypeface(MainApplication.getClimaconFont());
        txt.setText(getWeatherCode(f.getIcon()));
        txt.setTextSize(size);
    }

    public static String getWeatherCode(String icon) {
        icon = icon.replaceAll("nt_", "");
        if (icon.equals("cloudy")) {
            return Character.toString((char) 0x049);
        } else if (icon.equals("partlycloudy")) {
            return Character.toString((char) 0x022);
        } else
            return Character.toString((char) 0x049);
    }
}

File: app/src/main/java/fr/ekito/myweatherapp/WeatherFormat.kt
package fr.ekito.myweatherapp

import android.widget.TextView
import fr.ekito.myweatherlibrary.json.geocode.Geocode
import fr.ekito.myweatherlibrary.json.geocode.Location
import fr.ekito.myweatherlibrary.json.weather.Forecastday_

/**
 * Created by arnaud on 04/08/2016.
 */
object WeatherFormat {

    fun getLocation(geocode: Geocode): Location? {
        return geocode.results.first().geometry.location
    }

    fun weatherTxt(forecastday: Forecastday_): String {
        return forecastday.conditions //.getTitle() + "\n" + forecastday.getFcttextMetric();
    }

    fun filterForecast(forecastday: List<Forecastday_>): List<Forecastday_> {
        return forecastday.filter { f -> !f.icon.startsWith("nt_") }
                .take(4)
    }

    fun getTemp(day: Forecastday_): String {
        return day.low.celsius + "°C - " + day.high.celsius + "°C"
    }

    fun displayWeatherIcon(txt: TextView, size: Int, f: Forecastday_) {
        txt.typeface = MainApplication.climaconFont
        txt.text = getWeatherCode(f.icon)
        txt.textSize = size.toFloat()
    }

    fun getWeatherCode(icon: String): String {
        val icon = icon.replace("nt_".toRegex(), "")
        when (icon) {
            "cloudy" -> return Character.toString(0x049.toChar())
            "partlycloudy" -> return Character.toString(0x022.toChar())
            else -> return Character.toString(0x049.toChar())
        }
    }
}
